src/Controls/src/Core/Embedding/EmbeddingExtensions.cs

```diff
@@ -0,0 +1,122 @@
+﻿#if ANDROID || IOS || MACCATALYST || WINDOWS
+using System;
+using System.Linq;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Maui.Controls.Hosting;
+using Microsoft.Maui.Embedding;
+using Microsoft.Maui.Hosting;
+
+#if ANDROID
+using PlatformView = Android.Views.View;
+using PlatformWindow = Android.App.Activity;
+#elif IOS || MACCATALYST
+using PlatformView = UIKit.UIView;
+using PlatformWindow = UIKit.UIWindow;
+#elif WINDOWS
+using PlatformView = Microsoft.UI.Xaml.FrameworkElement;
+using PlatformWindow = Microsoft.UI.Xaml.Window;
+#endif
+
+namespace Microsoft.Maui.Controls.Embedding;
+
+/// <summary>
+/// A set of extension methods that allow for embedding a MAUI view within a native application.
+/// </summary>
+internal static class EmbeddingExtensions
```

Comment:
This class is internal for now since this is a new API and we cannot add APIs to net8.0. However, we can make this public in net9.0.

This API is added here to make development easier as well as to allow for testing/usage in the Community Toolkit (if they want). Instead of trying to do complex things from the outside, we can make it internal and the `[InternalsVisibleTo]` attributes effectively expose all these to them.